infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -723,79 +713,6 @@ private LogData readRecord(SegmentHandle segment, long address) throws IOExcepti
         }
     }
 
-    @Nullable
```

Comment:
In `private void readAddressSpace(SegmentHandle segment) throws` can you load the address space from the read channel and not the write channel ?

Reply:
the channel is used for reads and writes, I wouldn't like to touch it in this pr

Reply:
That's the point, the read channel opens the file in read only mode, so writes will fail. During the load, only reads should be permitted. Its more restrictive.

Reply:
But it works now somehow, let's create another PR for that